core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -252,12 +250,9 @@ public static CarbonTable buildTable(
       String tableName,
       Configuration configuration) throws IOException {
     TableInfo tableInfoInfer = CarbonUtil.buildDummyTableInfo(tablePath, "null", "null");
-    CarbonFile carbonFile = getLatestIndexFile(FileFactory.getCarbonFile(tablePath, configuration));
-    if (carbonFile == null) {
-      throw new RuntimeException("Carbon index file not exists.");
-    }
-    org.apache.carbondata.format.TableInfo tableInfo = CarbonUtil
-        .inferSchemaFromIndexFile(carbonFile.getPath(), tableName);
+    // InferSchema from data file
+    org.apache.carbondata.format.TableInfo tableInfo =
+        CarbonUtil.inferSchema(tablePath, tableName, false, configuration);
```

Comment:
Did you consider the performance for reading data from S3? it will increase the cost if inferSchema from S3 data file, right?

Reply:
only footer is read. so not much difference and only one IO. so, not much difference.
Index file was introduced to avoid multiple data file footer reading, but here for carbon reader to infer schema one file footer reading is enough.

Reply:
Carbon already support download little part data of whole file from S3?

Reply:
I think it's supported by default. Let me test once in S3

Reply:
@xubo245 : I have tested with huge data file (800 MB) and small index fie (500 KB), infer schema time is same for both these files.
reason is only header is read from s3 and header is in top itself, so no seek operations also.